delorean/interface.py

```diff
@@ -10,23 +10,47 @@
 UTC = "UTC"
 utc = timezone("utc")
 
+NAIVEMODE_UTC = "utc"
+NAIVEMODE_NAIVE = "naive"
+NAIVEMODE_RAISE = "raise"
+NAIVE_MODES = (NAIVEMODE_UTC, NAIVEMODE_NAIVE, NAIVEMODE_RAISE)
 
-def parse(s, dayfirst=True, yearfirst=True):
+def parse(s, dayfirst=True, yearfirst=True, naivemode=NAIVEMODE_UTC):
     """
     Parses a datetime string in it and returns a `Delorean` object.
 
     If a timezone is detected in the datetime string it will be
     normalized to UTC, and a Delorean object with that datetime and
     timezone will be returned.
+
+    If a timezone is not detected, the resulting timezone (and return object)
+    depends on the value of the naivemode argument:
+        '{mode_utc}' -- UTC is assumed
+        '{mode_raise}' -- An exception is raised
+        '{mode_naive}' -- Returns a naive datetime object (NOT a `Delorean`).
+
+    Note that in the case of naivemode=='{mode_naive}' a Delorean object is
+    not returned. This is because Delorean objects should always be timezone
+    aware.
+
     """
+
+    if not naivemode in NAIVE_MODES:
+        raise ValueError("Unknown naivemode value: '%s'" % naivemode)
+
     try:
         dt = capture(s, dayfirst=dayfirst, yearfirst=yearfirst)
     except:
         # raise a parsing error.
         raise ValueError("Unknown string format")
     if dt.tzinfo is None:
-        # assuming datetime object passed in is UTC
-        do = Delorean(datetime=dt, timezone=UTC)
+        if naivemode == NAIVEMODE_UTC:
+            # assuming datetime object passed in is UTC
+            do = Delorean(datetime=dt, timezone=UTC)
+        elif naivemode == NAIVEMODE_RAISE:
+            raise DeloreanInvalidDatetime("Unable to determine timezone info")
+        elif naivemode == NAIVEMODE_NAIVE:
+            do = dt #note that we are *not* returning a Delorean here
```

Comment:
The fact that it returns naive datetime obj sometimes and other times it returns `delorean` objects.